f600233d1b154d507b782f84604bcb12/update/sys_dictionary_x_g_dis_atat_contacts_formal_name.xml

```diff
@@ -0,0 +1,130 @@
+<?xml version="1.0" encoding="UTF-8"?><record_update>
+    <sys_dictionary action="INSERT_OR_UPDATE" element="formal_name" table="x_g_dis_atat_contacts">
+        <active>true</active>
+        <array>false</array>
+        <array_denormalized>false</array_denormalized>
+        <attributes/>
+        <audit>false</audit>
+        <calculation><![CDATA[// build rank or salutation (role dependent)
+function buildRankOrSalutation(current) {
+    var output = '';
+    //include rank for MIL
+    if (current.role == 'MILITARY') {
+        output += current.rank_components.abbreviation;
+        output += ' ';
+        //include salutation for CIV and CTR (if provided)
+    } else {
+        if (current.salutation != '') {
+            output += current.salutation.getDisplayValue();
+            output += ' ';
+        }
+    }
+    return output;
+}
+
+//build middle name (if provided)
+function buildMiddleName(current) {
+    var output = '';
+    if (current.middle_name != '') {
+        output += ' ';
+        output += current.middle_name;
+    }
+    return output;
+}
+
+//build suffix (if provided)
+//normalize common abbreviations Jr and Sr
+//jr/jr./Jr/Jr./JR/JR. should be normalized to Jr.
+//sr/sr./Sr/Sr./SR/SR. should be normalized to Sr.
+function buildSuffix(current) {
+    var output = '';
+    if (current.suffix != '') {
+        output += ' ';
+        //normalize
+        var isJuniorOrSenior = current.suffix.match(/^[JS]R[.]?$/i);
+        if (isJuniorOrSenior != null) {
+            output += current.suffix.charAt(0).toUpperCase() + current.suffix.slice(1).toLowerCase();
+            if (current.suffix.slice(-1) != '.') {
+                output += '.';
+            }
+            //include as is
+        } else {
+            output += current.suffix;
+        }
+    }
+    return output;
+}
+
+(function calculatedFieldValue(current) {
+    var output = '';
+    output += buildRankOrSalutation(current);
+    output += current.first_name; //mandatory
+    output += buildMiddleName(current);
+    output += ' ';
+    output += current.last_name; //mandatory
+    output += buildSuffix(current);
```

Comment:
Nice helper functions. Just a thought, but you might been able to cut back on some of the dot walking in the `buildMiddleName` and `buildSuffix` by passing in the value only rather than passing in `current`.